src/internal/rest.rs

```diff
@@ -428,6 +885,65 @@ impl<'a> IronCoreRequest<'a> {
         }
     }
 }
+// brought this private function in from reqwest
+fn replace_headers(dst: &mut HeaderMap, src: HeaderMap) {
+    // IntoIter of HeaderMap yields (Option<HeaderName>, HeaderValue).
+    // The first time a name is yielded, it will be Some(name), and if
+    // there are more values with the same name, the next yield will be
+    // None.
+    //
+    // TODO: a complex exercise would be to optimize this to only
+    // require 1 hash/lookup of the key, but doing something fancy
+    // with header::Entry...
```

Comment:
This doesn't look like it replaces anything. It adds headers to the destination list (apparently assuming they don't exist already), aggregating the values (if that's what `HeaderMap.append` does). Looks a little fishy.

Reply:
This code is lifted directly from `reqwest` is is what the `RequestBuilder` we were previously using was doing. (Now we build our own request so we can get things out for signing). I think the "replace" in the name is supposed to mean that if the header previously existed in the map, it will be overwritten.

An alternative would be to stop passing around individual headers as `HeaderMap`s and use a key value pair or some other abstraction.

Reply:
OK - I didn't understand how the `HeaderMap.insert` and `HeaderMap.append` worked. I guess this is reasonable.